pygmt/base_plotting.py

```diff
@@ -530,3 +535,105 @@ def image(self, imagefile, **kwargs):
             lib.call_module("image", arg_str)
+
+    @fmt_docstring
+    @use_alias(R="region", J="projection")
+    @kwargs_to_strings(
+        R="sequence",
+        angle="sequence_comma",
+        font="sequence_comma",
+        justify="sequence_comma",
+    )
+    def text(
+        self,
+        textfile=None,
+        x=None,
+        y=None,
+        text=None,
+        angle=None,
+        font=None,
+        justify=None,
+        **kwargs,
+    ):
+        """
+        Places text on a map.
+
+        Used to be pstext.
+
+        Takes in a textfile or (x,y,text) triples as input.
+
+        Must provide at least *textfile* or *x*, *y*, and *text*.
+
+        Full option list at :gmt-docs:`text.html`
+
+        {aliases}
+
+        Parameters
+        ----------
+        textfile : str
+            A text data file name
+        x, y : float or 1d arrays
+            The x and y coordinates, or an array of x and y coordinates to plot the text
+        text : str or 1d array
+            The text string, or an array of strings to plot on the figure
+        angle: int/float or bool
+            Set the angle measured in degrees counter-clockwise from horizontal. E.g. 30
+            sets the text at 30 degrees. If no angle is given then the input textfile
+            must have this as a column.
+        font : str or bool
+            Set the font specification with format "size,font,color" where size is text
+            size in points, font is the font to use, and color sets the font color. E.g.
+            "12p,Helvetica-Bold,red" selects a 12p red Helvetica-Bold font. If no font
+            info is given then the input textfile must have this information in one of
+            its columns.
+        justify: str or bool
+            Set the alignment which refers to the part of the text string that will be
+            mapped onto the (x,y) point. Choose a 2 character combination of L, C, R
+            (for left, center, or right) and T, M, B for top, middle, or bottom. E.g.,
+            BL for lower left. If no justification is given then the input textfile must
+            have this as a column.
+        {J}
+        {R}
+        """
+        kwargs = self._preprocess(**kwargs)
+
+        kind = data_kind(textfile, x, y, text)
+        if kind == "vectors" and text is None:
+            raise GMTInvalidInput("Must provide text with x and y.")
+
+        if angle is not None or font is not None or justify is not None:
+            if "F" not in kwargs.keys():
+                kwargs.update({"F": ""})
+            if angle is not None and isinstance(angle, (int, float)):
+                kwargs["F"] += f"+a{str(angle)}"
+            if font is not None and isinstance(font, str):
+                kwargs["F"] += f"+f{font}"
+            if justify is not None and isinstance(justify, str):
+                kwargs["F"] += f"+j{justify}"
+
+        with GMTTempFile(suffix=".txt") as tmpfile:
```

Comment:
We should look into passing text though virtual files, actually. I was avoiding this a while ago because the API had changed but that has been standardized now in GMT.

Reply:
Sorry, I don't quite get your intention here? Do you mean passing `text` (as in the command) through virtual files or 'text' (as in plain old text)? Either way I'm confused...